covid19-dashboard/dc-covid19/src/App.tsx

```diff
@@ -14,186 +14,122 @@
-                block: "nearest"
-            })
-        }
-    }
-
-    /**
-     * In charge of changing the state when the region, date or showTopN are updated.
-     * @param key
-     * @param value
-     */
-    handleSelectUpdate = (key: string, value: any) =>{
-        switch(key) {
-            case 'region':
-                this.setState({selectedRegion: value})
-                break;
-            case 'date':
-                this.setState({datePicked: value})
-                break;
-            case 'showTopN':
-                this.setState({selectedShowTopN: value})
-                break;
-        }
-    }
+/**
+ * Given a date and an array of deltaDays compute the real ISODates.
+ * For example, date = 2020-01-10 and deltaDays = [0, 1, 7] will return an array with the ISODates
+ * [2020-01-10, 2020-01-09, 2020-01-03] because 0 represents today, 1 represents 1 day ago, and 7 represents 7 days ago.
+ * @param date
+ * @param deltaDays
+ */
+const getRealISODatesFromArrayOfDeltaDays = (
+  date: string,
+  deltaDays: number[]
+): string[] => {
+  if (date === undefined || date === null) return [];
+  const dates = deltaDays.map(deltaDate =>
+    addOrSubtractNDaysToDate(date, -deltaDate)
+  );
+  return dates.sort((a, b) => (a < b ? -1 : a > b ? 1 : 0));
+};
 
-    /**
-     * Request the data from the server, and store the Promises in that order.
-     * The order is important [cases, deaths, population, places].
-     */
-    fetchData = () => {
-        const url: string = `/api/`
-        const apis = [
-            url + 'total-cases',
-            url + 'total-deaths',
-            url + 'population',
-            url + 'places'];
-        // Map every url to the promise of the fetch
-        return apis.map(api => fetch(api));
-    }
+/**
+ * Converts a number to a string including commas for readability.
+ * For example, int(10000) would get converted to str(10,000)
+ * @param num: the number to replace
+ */
+const numberWithCommas = (num: number): string => {
+  if (!num) return '0';
+  return num.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
+};
 
-    /**
-     * Returns only the data corresponding to a specific region. County, State or within a State.
-     * @param allData: all the data in the form of [cases, deaths, population, names]
-     * @param region: can be State, County or a State's geoId: geoId/XX
-     */
-    getStatsDataOnlyForRegion = (allData: {}[], region: string): {}[]=> {
-        let output: {}[] = []
-        const geoIdsContainedInSelectedRegion = filterGeoIdThatBelongTo(allData[3], region)
+/**
+ * Converts an ISO date to English date.
+ * For example, 2020-01-01 is converted to January 1st, 2020.
+ * @param ISOdate: a string representing an ISO date.
+ */
+const prettifyDate = (ISOdate: string): string => {
+  const prettifiedDate = moment(ISOdate).format('MMMM Do, YYYY');
+  // If prettifiedDate is not valid, it means it wasn't a ISOdate input.
+  if (!prettifiedDate) return 'Some Day';
```

Comment:
if not valid, should return ISOdate, btw, i think the variable name should be "isoDate"